sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -1042,6 +1042,47 @@ struct _pi_queue : _pi_object {
 
   // Indicates that the queue is healthy and all operations on it are OK.
   bool Healthy{true};
+
+  // Data structures and methods for optimizing events usage in in-order queues.
+
+  // Copy of the last command event which is suitable for reuse.
+  // It will be put into the cache when new command is submitted to the in-order
+  // queue.
+  pi_event CandidateForReuse = nullptr;
+
+  // Caches of events for reuse.
+  std::vector<std::list<pi_event>> EventCaches{2};
```

Comment:
I don't understand why we need a per-queue cache of events.

Reply:
To be able to reuse event across queues we need to wait when event is completed and its external references are gone which almost always means that event is ready to be destroyed -> we already have context-level cache of events for that.

For in-order queue we can reuse event much earlier, We don't need to wait for its completion,  we can rely on dependency chain between commands in the in-order queue.
This dependency chain is not maintained across different queues, it is maintained only in scope of in-order queue. That's why we need to have cache of events in the queue level.